sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,355 @@
+                    else:
+                        msg = "invalid GrimoireLab yaml format. List of elements expected for organization %s" % name
+                        raise InvalidFormatError(cause=msg)
+                self._organizations[name] = o
+
+        except KeyError as e:
+            msg = "invalid GrimoireLab yaml format. Attribute %s not found" % e.args
+            raise InvalidFormatError(cause=msg)
+
+        except TypeError as e:
+            msg = "invalid GrimoireLab yaml format. %s" % e.args
+            raise InvalidFormatError(cause=msg)
+
+    def __parse_affiliations_yml(self, affiliations, uuid):
+        """Parse identity's affiliations from a yaml dict."""
+        enrollments = []
+
+        for aff in affiliations:
+            name = self.__encode(aff['organization'])
+            if not name:
+                msg = "invalid GrimoireLab yaml format. Empty organization name"
+                raise InvalidFormatError(cause=msg)
+
+            # we trust the Organization name included in the identities file
+            org = Organization(name=name)
+
+            if org is None:
+                continue
+
+            if 'start' in aff:
+                start_date = self.__force_datetime(aff['start'])
+            else:
+                start_date = MIN_PERIOD_DATE
+
+            if 'end' in aff:
+                end_date = self.__force_datetime(aff['end'])
+            else:
+                end_date = MAX_PERIOD_DATE
+
+            enrollment = Enrollment(start=start_date, end=end_date,
+                                    organization=org)
+            enrollments.append(enrollment)
+
+        self.__validate_enrollment_periods(enrollments)
+
+        return enrollments
+
+    def __force_datetime(self, obj):
+        """Converts ojb to time.datetime.datetime
+
+        YAML parsing returns either date or datetime object depending
+        on how the date is written. YYYY-MM-DD will return a date and
+        YYYY-MM-DDThh:mm:ss will return a datetime
+
+        :param obj: date or datetime object
+        """
+        if isinstance(obj,datetime.datetime):
+            return obj
+
+        t = datetime.time(0,0)
+        return datetime.datetime.combine(obj, t)
+
+    def __load_yml(self, stream):
+        """Load yml stream into a dict object """
+        try:
+            return yaml.load(stream)
+        except ValueError as e:
+            cause = "invalid yml format. %s" % str(e)
+            raise InvalidFormatError(cause=cause)
+
+    def __encode(self, s):
+        import sys
+
+        if sys.version_info[0] >= 3: # Python 3
+            return s if s else None
+        else: # Python 2
+            if type(s) is str:
+                return s.encode('UTF-8') if s else None
+            else:
+                return s
+
+    def __compose_uuid(self, name, email, username):
+        """Composes a uuid string as result of the concatenation of name and (email and/or username)"""
```

Comment:
Why do we need this uuid? These uuids are ignored by sortinghat when it loads the file.